tests/tls_apis/enc/enc.cpp

```diff
@@ -0,0 +1,337 @@
+// accept a connection reqest
+oe_result_t enclave_identity_verifier(oe_identity_t* identity, void* arg)
+{
+    oe_result_t result = OE_VERIFY_FAILED;
+
+    (void)arg;
+    OE_TRACE_INFO("enclave_identity_verifier is called with parsed report:\n");
+
+    // Check the enclave's security version
+    if (identity->security_version < 1)
+    {
+        OE_TRACE_ERROR(
+            "identity->security_version checking failed (%d)\n",
+            identity->security_version);
+        goto done;
+    }
+
+    // Dump an enclave's unique ID, signer ID and Product ID. They are
+    // MRENCLAVE, MRSIGNER and ISVPRODID for SGX enclaves. In a real scenario,
+    // custom id checking should be done here
+
+    OE_TRACE_INFO("identity->signer_id :\n");
+    for (int i = 0; i < OE_UNIQUE_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->signer_id[i]);
+
+    OE_TRACE_INFO("\nparsed_report->identity.signer_id :\n");
+    for (int i = 0; i < OE_SIGNER_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->signer_id[i]);
+
+    OE_TRACE_INFO("\nidentity->product_id :\n");
+    for (int i = 0; i < OE_PRODUCT_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->product_id[i]);
+
+    result = OE_OK;
+done:
+    return result;
+}
+
+// input: input_data and input_data_len
+// output: key, key_size
+oe_result_t generate_key_pair(
+    int key_type,
+    uint8_t** public_key,
+    size_t* public_key_size,
+    uint8_t** private_key,
+    size_t* private_key_size)
+{
+    oe_result_t result = OE_FAILURE;
+    oe_asymmetric_key_params_t params;
+    char user_data[] = "test user data!";
+    size_t user_data_size = sizeof(user_data) - 1;
+    uint8_t* local_public_key = NULL;
+    uint8_t* local_private_key = NULL;
+
+    OE_TRACE_INFO("Generate key pair");
+
+    if (key_type == MBEDTLS_PK_ECKEY)
+    {
+        params.type =
+            OE_ASYMMETRIC_KEY_EC_SECP256P1; // MBEDTLS_ECP_DP_SECP256R1
+        params.format = OE_ASYMMETRIC_KEY_PEM;
+        params.user_data = user_data;
+        params.user_data_size = user_data_size;
+        result = oe_get_public_key_by_policy(
+            OE_SEAL_POLICY_UNIQUE,
+            &params,
+            public_key,
+            public_key_size,
+            NULL,
+            NULL);
+        OE_CHECK(result);
+
+        result = oe_get_private_key_by_policy(
+            OE_SEAL_POLICY_UNIQUE,
+            &params,
+            private_key,
+            private_key_size,
+            NULL,
+            NULL);
+        OE_CHECK(result);
+    }
+    else if (key_type == MBEDTLS_PK_RSA)
+    {
+        int res = -1;
+        mbedtls_ctr_drbg_context ctr_drbg_contex;
+        mbedtls_entropy_context entropy_context;
+        mbedtls_pk_context pk_context;
+        size_t local_public_key_size = 512;
+        size_t local_private_key_size = 2048;
+
+        mbedtls_ctr_drbg_init(&ctr_drbg_contex);
+        mbedtls_entropy_init(&entropy_context);
+        mbedtls_pk_init(&pk_context);
```

Comment:
This doesn't get freed with `mbedtls_pk_free`, which causes the memory leak in CI.

Reply:
good catch! Thanks.
Fixed